env0/data_project.go

```diff
@@ -162,14 +182,64 @@ func getProjectByName(name string, parentId string, parentName string, meta inte
 	return projectsByName[0], nil
 }
 
+func pathMatches(path, parentIds []string, meta interface{}) (bool, error) {
+	if len(path) != len(parentIds) {
+		return false, nil
+	}
+
+	apiClient := meta.(client.ApiClientInterface)
+
+	for i := range path {
+		parentId := parentIds[i]
+
+		parentProject, err := apiClient.Project(parentId)
+		if err != nil {
+			return false, fmt.Errorf("failed to get a parent project with id '%s': %w", parentId, err)
+		}
+
+		if parentProject.Name != path[i] {
+			return false, nil
+		}
+	}
+
+	return true, nil
+}
+
+func filterByParentProjectPath(parentPath string, projectsByName []client.Project, meta interface{}) ([]client.Project, error) {
+	filteredProjects := make([]client.Project, 0)
+
+	path := strings.Split(parentPath, "|")
+
+	for _, project := range projectsByName {
+		parentIds := strings.Split(project.Hierarchy, "|")
+		// right most element is the project itself, remove it.
+		parentIds = parentIds[:len(parentIds)-1]
+
+		matches, err := pathMatches(path, parentIds, meta)
```

Comment:
mmmm that feels a bit complicated, couldn't we have filtered the projects by their hierarchy seeing if `hierarchy.contains(searchedPath)` ?
Or am I missing something?

Reply:
oh oh nvm I am wrong since those are IDs not names so it won't work that way 👌